docs/reference/modules/cluster/disk_allocator.asciidoc

```diff
@@ -75,13 +75,25 @@ Defaults to `true`. Set to `false` to disable the disk allocation decider. Upon
 Controls the low watermark for disk usage. It defaults to `85%`, meaning that {es} will not allocate shards to nodes that have more than 85% disk used. It can alternatively be set to a ratio value, e.g., `0.85`. It can also be set to an absolute byte value (like `500mb`) to prevent {es} from allocating shards if less than the specified amount of space is available. This setting has no effect on the primary shards of newly-created indices but will prevent their replicas from being allocated.
 // end::cluster-routing-watermark-low-tag[]
 
+`cluster.routing.allocation.disk.watermark.low.max_headroom`::
+(<<dynamic-cluster-setting,Dynamic>>) Controls the max headroom for the low watermark (in case of a percentage/ratio value).
+Defaults to 200GB when `cluster.routing.allocation.disk.watermark.low` is not explicitly set.
+This caps the amount of free space required.
+If the low watermark max headroom is set, the `cluster.routing.allocation.disk.watermark.high.max_headroom` needs to be set as well.
```

Comment:
I am not sure this is true? I think the only constraint we have is on the size of the max headroom? I.e., if low head room is set, it must be larger than the high head room value (which can be the default value or an explicit value)?

I would only consider it "set" if explicitly set and not using the default value.

Reply:
@henningandersen it only applies in the rare corner cases where one sets e.g. the corresponding watermark percentage, rending the headroom to being not set (-1). This is exemplified in tests org.elasticsearch.cluster.routing.allocation.DiskThresholdSettingsTests#testInvalidLowHeadroomSetAndHighNotSet and testInvalidHighHeadroomSetAndFloodNotSet. But they are rather rare cases. I understand it gives you uneasiness in reading this, so I can remove it, to simplify the documentation, and just let the users be guided by the corresponding error message in case they fall in the corner case. So I will go ahead and simply remove this line (also the similar line for the high headroom paragraph).